### PR TITLE
remove iris-grib dependency

### PR DESCRIPTION
**Pull request by marqh**
iris-grib ought not to be an iris dependency, as it is not available across platforms and python version

### CODE REVIEW

**Comment by marqh** [comment]
hmmm, the circleci has not triggered

I've reproduced this on another PR

not sure why this would be

**Comment by ocefpaf** [comment]
@marqh you will need to navigate to https://circleci.com/gh/marqh/conda-recipes-scitools/tree/master and click on 'follow project' to enable `CircleCI` in your fork. Then submit an empty commit here and the test should start.

That is the new and "improved" `CircleCi` behavior :unamused:

**Comment by marqh** [comment]
> That is the new and "improved" CircleCi behavior 😒

cheers for the tip @ocefpaf easy enough to fix up

however the build is now failing :(

**Comment by marqh** [comment]
hmmm

```
===== testing package: iris-1.11.0-np110py34_1 =====
import: 'iris'
import: 'iris.fileformats.pp_packing'
Traceback (most recent call last):
  File "/opt/conda/conda-bld/iris_1483458539302/test_tmp/run_test.py", line 30, in <module>
    import iris.fileformats.pp_packing
  File "/opt/conda/conda-bld/iris_1483458539302/_t_env/lib/python3.4/site-packages/iris/fileformats/pp_packing.py", line 32, in <module>
    from iris.fileformats import _old_pp_packing as old_pp_packing
ImportError: dynamic module does not define init function (PyInit__old_pp_packing)
TESTS FAILED: iris-1.11.0-np110py34_1
```

it's an actual error, not just a build artifact

perhaps this is worth chasing down, but perhaps it is time to accpet that mostly noone is using the built in packing, if they install via conda they get mo_pack to handle this

this would simplify the build matrix as well, as there would be no numpy version dependency left for iris

**Comment by marqh** [comment]
To restate: in this PR I have removed Iris' dependency on Iris-GRIB and removed Iris' pp_packing.so build.  I have also reinstated the iris test data, which was removed years ago due to it's huge sie, which has now been greatly reduced.

I think this represents a reasonable  set of build decisions, but they are all changes from previous behaviour.  Merging this would gain my support (I put it up after all), but it should be done with these changes in mind, there will be implications.

cheers
mark

**Comment by pp-mo** [comment]
>  I have also reinstated the iris test data

@marqh I'm confused by this : I don't see such a change here ?

**Comment by marqh** [comment]
@pp-mo 
> I'm confused by this : I don't see such a change here ?

i removed this line

https://github.com/SciTools/conda-recipes-scitools/pull/222/files?diff=unified#diff-dd7a8a2d2ab1f4e3ec87088874ddcc02L16

**Comment by pp-mo** [comment]
I've built this, installed in a repo and run the tests.

( Aside:  I found that ```python -m unittest discover .../lib/iris.tests``` doesn't quite work, because it wants "gribapi".
This is because the mechanism for skipping all the older tests under ```iris.tests.unit.fileformats.grib``` is in ```iris.tests.runner```, not in the tests themselves. 
However, you can use ```python -c "import iris.tests.runner.__main__``` to run it correctly.
)

**Remaining problems**:
The following packages are also needed to make all the test pass :
 * pep8
 * filelock
 * imagehash

Note : these requirements *are* all included in iris/conda-requirements.txt

Additional packages (sphinx, iris_sample_data) are still needed to build docs.
I think that's ok.

**Comment by pp-mo** [comment]
>  pep8, filelock, imagehash

** @marqh would you like add these ones to ```meta.yaml``` ?**

All tests will then pass on an installed Iris (subject to running them 'the right way').

**Comment by marqh** [comment]
ping @pp-mo 
i think this is working now

**Comment by dkillick** [comment]
@marqh this is missing an AppVeyor build - please can you address as appropriate.

**Comment by dkillick** [comment]
@marqh could you please comment on the changed version of conda?

**Comment by marqh** [comment]
> @marqh could you please comment on the changed version of conda?

in
https://circleci.com/gh/marqh/conda-recipes-scitools/9?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
https://ci.appveyor.com/project/pelson/conda-recipes-scitools-311/build/1.0.682/job/gtc29yr48jjrgwg3
https://travis-ci.org/SciTools/conda-recipes-scitools/builds/192343603

the following new condaism arrives
```
Fetching package metadata ...........

Resolving distributions from 33 recipes... 

Traceback (most recent call last):

  File "/Users/travis/miniconda/bin/conda-build-all", line 6, in <module>

    sys.exit(conda_build_all.cli.main())

  File "/Users/travis/miniconda/lib/python3.5/site-packages/conda_build_all/cli.py", line 90, in main

    b.main()

  File "/Users/travis/miniconda/lib/python3.5/site-packages/conda_build_all/builder.py", line 255, in main

    all_distros = self.compute_build_distros(index, recipe_metas, build_config)

  File "/Users/travis/miniconda/lib/python3.5/site-packages/conda_build_all/builder.py", line 220, in compute_build_distros

    self.matrix_conditions)

  File "/Users/travis/miniconda/lib/python3.5/site-packages/conda_build_all/resolved_distribution.py", line 83, in resolve_all

    cases = sorted(vn_matrix.special_case_version_matrix(meta, index))

  File "/Users/travis/miniconda/lib/python3.5/site-packages/conda_build_all/version_matrix.py", line 212, in special_case_version_matrix

    py_vn = minor_vn(index[python_pkg.fn]['version'])

AttributeError: 'Dist' object has no attribute 'fn'
```

I am not aware of conda_build_all enough to immediately see what `python_pkg` is and why it used to have a `fn` that it no longer has

i have pinned back to the last working version of conda and (after a stutter or two) seen that this enables building

**Comment by marqh** [comment]
the fixes in this to manage (pin) the conda version are now critical, as our master builds are all failing.
a merge on this PR would be most appreciated

**Comment by pp-mo** [comment]
This PR has now escalated from just updating the iris recipe to also changing several others:
```
(p27_irisdeps) [itpp@eld238 conda-recipes-scitools]$ git diff --stat=165,100 85c3fa8~1
 .travis.yml                 |    4 ++--
 appveyor.yml                |    4 ++--
 imagehash/bld.bat           |    1 +
 imagehash/meta.yaml         |    3 ++-
 iris/bld.bat                |    3 ---
 iris/build.sh               |    4 +---
 iris/meta.yaml              |    9 ++++-----
 olefile/meta.yaml           |   26 ++++++++++++++++++++++++++
 pywavelets/meta.yaml        |    1 -
 scripts/run_docker_build.sh |    4 +++-
 10 files changed, 41 insertions(+), 18 deletions(-)
(p27_irisdeps) [itpp@eld238 conda-recipes-scitools]$ 
```

So, I thought I'd go around the loop of building these from recipes and creating a test environment.
Thus I have tried to build imagehash, olefile + pywavelets in addition to iris.
Unfortunately, I'm now getting an error with imagehash:
```
> conda build ./imagehash --numpy 1.10
 . . .
RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.python.org/simple/olefile/r
Command failed: /bin/bash -x -e /data/local/itpp/miniconda/conda-bld/imagehash_1484733994681/work/ImageHash-3.1/conda_build.sh, output: Command '['/bin/bash', '-x', '-e', '/data/local/itpp/miniconda/conda-bld/imagehash_1484733994681/work/ImageHash-3.1/conda_build.sh']' returned non-zero exit status 1
```

It's possible I'm not doing the right thing here though.
Maybe I am running with bad conda tools versions?
I currently have conda 4.3.4 and conda-build 2.1.1 (was 2.1.0)
Is that maybe the real problem here ?

**Comment by marqh** [comment]
@pp-mo please may you check the circleci log and align your versions.  perhaps you can use the scripts/dockerscript
(probably from here https://github.com/SciTools/conda-recipes-scitools/blob/master/scripts/run_docker_build.sh#L37)

if this is an issue, i don't understand why it isn't present in circleci builds

**Comment by dkillick** [comment]
> I currently have conda 4.3.4

@pp-mo I believe that @marqh had to pin the conda version in this PR to v4.2.13 precisely to avoid the problem you quoted.

**Comment by marqh** [comment]
@dkillick i think that is incorrect, in that the version of conda would not affect this problem of dependency resolution

**Comment by marqh** [comment]
@pp-mo is it possible that your build process is not respecting that olefile is not a required dependency on linux?

please could you remove the line marking olefile from the meta.yaml for imagehash and try to build that again?

thank you
mark

**Comment by marqh** [comment]
just to note, hashlib is not being rebuilt on linux in this PR, it is already in situ

**Comment by marqh** [comment]
jeepers, and the saga continues

i have reprodued the error you found @pp-mo i have tried to add a commit to fix this, we will see what the testing says

**Comment by marqh** [comment]
ok, circle is building now, though  travis and appveyor are very slow to update

i am confused.  to fix imagehash i have included a dependency on olefile on linux

however this wasn't needed before
https://github.com/SciTools/conda-recipes-scitools/pull/208
https://circleci.com/gh/SciTools/conda-recipes-scitools/446?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

changes are occurring here which I don't understand, I don't see why this change has seemingly been required.

I am content to take this as an improvement and investigate further from here (i.e. someone merge this)

I am out of action and off grid for a few days, so if someone want to make changes, may I suggest you adopt the diffs and work from there in a replacement PR

thank you

**Comment by marqh** [comment]
rebased following #231 

@dkillick please may we try to get this one merged next?

File: iris/meta.yaml [hunk 1]
```diff
@@ -36,20 +35,20 @@ requirements:
         - cartopy >=0.14
         - matplotlib
         - netcdf4
-        - ecmwf_grib >=1.12.1  # [not win]
         # There is a numpy C ABI dependency for as long as we have a C extension (pp_packing).
```
Comment (lbdreyer) [inline iris/meta.yaml]: @marqh If we are removing the pp packing we should remove this line as well.

File: iris/meta.yaml [hunk 2]
```diff
@@ -20,14 +20,13 @@ requirements:
         - biggus >=0.14
         - cartopy >=0.14
         - netcdf4
-        - numpy x.x
+        - numpy
```
Comment (dkillick) [inline iris/meta.yaml]: This (`numpy`) is not the same as `numpy x.x`. Are you intentionally adjusting the numpy requirement here?

File: iris/meta.yaml [hunk 3]
```diff
@@ -36,20 +35,19 @@ requirements:
         - cartopy >=0.14
         - matplotlib
         - netcdf4
-        - ecmwf_grib >=1.12.1  # [not win]
-        # There is a numpy C ABI dependency for as long as we have a C extension (pp_packing).
-        - numpy x.x
+        - numpy
```
Comment (dkillick) [inline iris/meta.yaml]: Again.

File: iris/meta.yaml (hunk 2 above)
Comment (marqh) [inline iris/meta.yaml]: yes, this is no longer required, as i have removed the pp_packing build step, which is obsolete and deprecated, replaced by mo_pack

**Comment by bjlittle** [comment]
:+1:

**Comment by bjlittle** [comment]
@marqh I'll wait for the CI tests to complete before mreging ...

**Pull request merged**